lib/ui/painting.dart

```diff
@@ -1607,12 +1655,152 @@ class Path extends NativeFieldWrapperClass2 {
   /// Returns a copy of the path with all the segments of every
   /// subpath transformed by the given matrix.
   Path transform(Float64List matrix4) {
-    assert(matrix4 != null);
-    if (matrix4.length != 16)
-      throw new ArgumentError('"matrix4" must have 16 entries.');
+    assert(_matrix4IsValid(matrix4));
     return _transform(matrix4);
   }
   Path _transform(Float64List matrix4) native 'Path_transform';
+
+  /// Gets the bounding rectangle for this path.
+  Rect getBounds() {
+    final Float32List rect = _getBounds();
+    return new Rect.fromLTRB(rect[0], rect[1], rect[2], rect[3]);
+  }
+  Float32List _getBounds() native 'Path_getBounds';
+
+  /// Sets this path to the result of operation [PathOperation]
+  /// 
+  /// If only one path is specified, this path will be used as the first 
+  /// operand, otherwise path1 will be the first operand and path2 the
+  /// second.
+  /// 
+  /// The resulting path will be constructed from non-overlapping contours. The
+  /// curve order is reduced where possible so that cubics may be turned into
+  /// quadratics, and quadratics maybe turned into lines.
+  static Path combine(PathOperation operation, Path path1, Path path2) {
+    assert(path1 != null);
+    assert(path2 != null);
+
+    final Path path = new Path();
+    if (path._op(path1, path2, operation.index)) {
+      return path;
+    } 
+    throw new StateError('Path combine failed - likely due to invalid path');
+  }
+  bool _op(Path path1, Path path2, int operation) native 'Path_op';
+
+  /// Creates a [PathMetrics] object for this path.
+  PathMetrics computeMetrics({bool forceClosed = false}) {
+    return new PathMetrics._(this, forceClosed);
+  }
+}
+
+/// Convenience class to return the result of [PathMetric.getTangentForOffset].
+class Tangent {
+  /// Position of the tangent
+  final Offset position;
+  /// Angle of the tangent in radians
```

Comment:
```
  /// The direction of the curve at [position].
  ///
  /// When used with [PathMetric.getTangentForOffset], this is the angle of the curve that is the given offset
  /// along the path (i.e. the direction of the curve at [position]).
  ///
  /// This value is in radians, with 0.0 meaning pointing along the x axis in the positive x-axis direction, and
  /// positive numbers going clockwise from there.
```

Check that that part about the meaning is correct, though. :-)

Reply:
Positive numbers make it point "up" towards the y axis, negative "down" - I think it's more visually helpful to reference the y-axis than to reference clock direction (counterclockwise for positive, clockwise for negative)

Reply:
Hm, we should flip the sign over to match arcTo (which is documented to work as I described here).

I'm fine with using the up/down terminology if you prefer. Maybe do both. :-)